### ISSUE TITLE
Ignored nodes in rule 'indent' does not completely ignore the node

### ISSUE THREAD

**Issue by gudoy** [eslint/eslint#10542]
* **ESLint Version:** 5.0.1
* **Node Version:** 10.3.0
* **npm Version:** 6.1.0

**What parser are you using?**
Default

**What did you do**

```js
class LanguageProvider extends Component {

    constructor(props) {
        console.log('LANGUAGEPROVIDER.constructor');

        super(props);

        // Get user preferred language
        const locale = i18n.getUserPreferedLanguage();

console.log('user preferered lang', locale);

        this.state = {
            locale,
            messages: {},
        };
    }
   // ...
}
```

**Eslint conf**
```js
"indent": [
            "warn",
            4,
            {
                "ignoreComments": true,
                "ignoredNodes": [
                    "ExpressionStatement[expression.callee.object.name=console]"
                ]
            }
        ],
}
```

**Launched commmand**
```bash
eslint app/src --ext .js --ext .jsx --color
```

**What did you expect to happen?**
I expected the whole console.log() line to be COMPLETELY ignored by the rule and not taken into account when setting the indentation level for the next lines
My goal is to allow lines with console.log to have an indentation of 0 (to be easily spotted and removed when dev is finished).

**What actually happened? **
Without the ignoreNodes option, I get warnings for console.log line with invalid indentation like:
35:1  warning  Expected indentation of 8 spaces but found 0  indent
84:1  warning  Expected indentation of 8 spaces but found 0  indent

With the option with the AST expression, the nodes no longer trigger a warning, but they indentation level is still taken into account as the new base for the next lines and this, I get the following warnings :
37:1  warning  Expected indentation of 0 spaces but found 8   indent
  38:1  warning  Expected indentation of 4 spaces but found 12  indent
  39:1  warning  Expected indentation of 4 spaces but found 12  indent
  40:1  warning  Expected indentation of 0 spaces but found 8   indent
  86:1  warning  Expected indentation of 0 spaces but found 8   indent

**Comment by platinumazure** [comment]
Seems like this might be a general bug in indent, where one incorrectly indented line could set the expected level for later lines (including if the line is supposed to be ignored).

@not-an-aardvark Any thoughts?

**Comment by not-an-aardvark** [comment]
I think this might be another case where things are technically working as designed but the design leads to a non-ideal outcome in some cases. It's happening as a result of two factors:

* When a node is "ignored", all of the nodes that depend on it still have their indentation checked relative to the node's actual location in the source code. (This is necessary so that people can ignore things like functions expressions while still checking statements inside the expressions.)
* When setting the indentation of lists of elements (e.g. parameters, arguments, statements in a block), the indentation of each element is set relative to the element before it, rather than to the start of the list. (I believe there's a good reason I did it this way, but I don't remember what it was. One way to find out would be to change the behavior and see what tests start failing.)
